code/datums/jobs/job/marines.dm

```diff
@@ -376,7 +376,7 @@ You are also in charge of communicating with command and letting them know about
 		return
 	if(!new_human.assigned_squad)
 		return
-	if(!new_human.assigned_squad.squad_leader)
+	if(!ismarineleaderjob(new_human.assigned_squad?.squad_leader?.job)) //If there's no proper SL already in the squad, promote to leader
```

Comment:
you're already early returning above if there is no assigned squad, don't need the ? for that bit.